### DISCUSSION TITLE
Use error codes for errors.

### DISCUSSION THREAD

**Discussion by PaulHariel** [Cumteam/Cumpilator#21] [Ideas]
Should we use error codes to handle errors? It would make the code execution flow more controllable and allow the toolchain to work on more platforms.

Example of error handling using exceptions:
```cpp
int fun(args) {
    int result;
    try {
        result = try_get_value();
    } catch (lib::value_exception e) {}
    // ...
    catch (std::exception e) { std::print(e.message()); return 0; }
    return result;
}
```

Example of error handling using error codes:
```cpp
int fun(args, lib::error_code& ec) noexcept {
    int result = get_value(ec);
    if (ec) {
        if (ec.value() == lib::error::value_error)
            std::print(ec.message());
        return 0;
    }
    return result;
}

**Comment by Mideky-hub** [comment]
Je préfère travailler avec des exceptions car ça demande moins d'implémentation dans l'usage direct de la fonction.
